chat/src/main/java/bisq/chat/ChatService.java

```diff
@@ -154,6 +155,8 @@ public CompletableFuture<Boolean> initialize() {
 
         list.add(chatNotificationService.initialize());
 
+        userIdentityService.setIsIdentityUsed(this::isIdentityUsed);
```

Comment:
Here is my suggestion:
```
Index: chat/src/main/java/bisq/chat/ChatService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/chat/src/main/java/bisq/chat/ChatService.java b/chat/src/main/java/bisq/chat/ChatService.java
--- a/chat/src/main/java/bisq/chat/ChatService.java	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/chat/src/main/java/bisq/chat/ChatService.java	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -39,6 +39,7 @@
 import bisq.security.pow.ProofOfWorkService;
 import bisq.settings.SettingsService;
 import bisq.user.UserService;
+import bisq.user.identity.UserIdentity;
 import bisq.user.identity.UserIdentityService;
 import bisq.user.profile.UserProfile;
 import bisq.user.profile.UserProfileService;
@@ -49,6 +50,7 @@
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Slf4j
 @Getter
@@ -205,7 +207,22 @@
             optionalChannel.ifPresent(channel -> getChatChannelSelectionService(chatChannelDomain).selectChannel(channel));
             return optionalChannel;
         }
+    }
 
+    public boolean isIdentityUsed(UserIdentity userIdentity) {
+        boolean usedInAnyPrivateChannel = Stream.concat(bisqEasyOpenTradeChannelService.getChannels().stream(),
+                        twoPartyPrivateChatChannelServices.values().stream()
+                                .flatMap(c -> c.getChannels().stream()))
+                .anyMatch(c -> c.getMyUserIdentity().equals(userIdentity));
+
+        boolean usedInAnyMessage = Stream.concat(bisqEasyOfferbookChannelService.getChannels().stream()
+                                .flatMap(c -> c.getChatMessages().stream()),
+                        commonPublicChatChannelServices.values().stream()
+                                .flatMap(c -> c.getChannels().stream())
+                                .flatMap(c -> c.getChatMessages().stream())
+                )
+                .anyMatch(m -> m.getAuthorUserProfileId().equals(userIdentity.getId()));
+        return usedInAnyPrivateChannel || usedInAnyMessage;
     }
 
     public ChatChannelSelectionService getChatChannelSelectionService(ChatChannelDomain chatChannelDomain) {
Index: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileController.java b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileController.java
--- a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileController.java	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileController.java	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -17,6 +17,7 @@
 
 package bisq.desktop.main.content.user.user_profile;
 
+import bisq.chat.ChatService;
 import bisq.common.observable.Pin;
 import bisq.desktop.ServiceProvider;
 import bisq.desktop.common.observable.FxBindings;
@@ -42,6 +43,7 @@
 import java.util.concurrent.CompletableFuture;
 
 import static bisq.desktop.common.view.NavigationTarget.CREATE_PROFILE_STEP1;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 @Slf4j
 public class UserProfileController implements Controller {
@@ -51,11 +53,13 @@
     private final UserIdentityService userIdentityService;
     private final ReputationService reputationService;
     private final ProfileAgeService profileAgeService;
+    private final ChatService chatService;
     private Pin userProfilesPin, selectedUserProfilePin, reputationChangedPin;
     private Subscription statementPin, termsPin;
 
 
     public UserProfileController(ServiceProvider serviceProvider) {
+        chatService = serviceProvider.getChatService();
         userIdentityService = serviceProvider.getUserService().getUserIdentityService();
         reputationService = serviceProvider.getUserService().getReputationService();
         profileAgeService = reputationService.getProfileAgeService();
@@ -71,6 +75,7 @@
 
         selectedUserProfilePin = FxBindings.subscribe(userIdentityService.getSelectedUserIdentityObservable(),
                 userIdentity -> {
+                    updateButtons();
                     if (userIdentity == null) {
                         return;
                     }
@@ -94,9 +99,14 @@
         reputationChangedPin = reputationService.getChangedUserProfileScore().addObserver(userProfileId -> UIThread.run(this::applyReputationScore));
 
         statementPin = EasyBind.subscribe(model.getStatement(),
-                statement -> updateSaveButtonState());
+                statement -> updateButtons());
         termsPin = EasyBind.subscribe(model.getTerms(),
-                terms -> updateSaveButtonState());
+                terms -> updateButtons());
+    }
+
+    private void updateButtons() {
+        updateSaveButtonState();
+        updateDeleteButtonState();
     }
 
     private void updateSaveButtonState() {
@@ -112,6 +122,17 @@
                 terms.equals(userProfile.getTerms()));
     }
 
+    private void updateDeleteButtonState() {
+        UserIdentity userIdentity = userIdentityService.getSelectedUserIdentity();
+        if (userIdentity == null) {
+            return;
+        }
+
+        boolean cannotDelete = chatService.isIdentityUsed(userIdentity) || !userIdentityService.hasMultipleUserIdentities();
+        model.getDeleteButtonDisabled().set(cannotDelete);
+        model.getUseDeleteTooltip().set(cannotDelete);
+    }
+
     @Override
     public void onDeactivate() {
         userProfilesPin.unbind();
@@ -146,25 +167,22 @@
                     UIThread.runOnNextRenderFrame(() -> {
                         UserIdentity value = userIdentityService.getSelectedUserIdentity();
                         model.getSelectedUserIdentity().set(value);
-                        updateSaveButtonState();
+                        updateButtons();
                     });
                 });
     }
 
-    public void onDelete() {
-        if (userIdentityService.getUserIdentities().size() < 2) {
-            new Popup().warning(Res.get("user.userProfile.deleteProfile.lastProfile.warning")).show();
-        } else {
-            new Popup().warning(Res.get("user.userProfile.deleteProfile.warning"))
-                    .onAction(this::doDelete)
-                    .actionButtonText(Res.get("user.userProfile.deleteProfile.warning.yes"))
-                    .closeButtonText(Res.get("action.cancel"))
-                    .show();
-        }
+    public void onDeleteProfile() {
+        String profileName = checkNotNull(userIdentityService.getSelectedUserIdentity()).getUserName();
+        new Popup().warning(Res.get("user.userProfile.deleteProfile.popup.warning", profileName))
+                .onAction(this::doDeleteProfile)
+                .actionButtonText(Res.get("user.userProfile.deleteProfile.popup.warning.yes"))
+                .closeButtonText(Res.get("action.cancel"))
+                .show();
     }
 
-    private CompletableFuture<DataService.BroadCastDataResult> doDelete() {
-        return userIdentityService.deleteUserProfile(userIdentityService.getSelectedUserIdentity())
+    private CompletableFuture<DataService.BroadCastDataResult> doDeleteProfile() {
+        return userIdentityService.deleteUserIdentity(userIdentityService.getSelectedUserIdentity())
                 .whenComplete((result, throwable) -> {
                     if (throwable != null) {
                         UIThread.run(() -> new Popup().error(throwable).show());
@@ -173,7 +191,7 @@
                             UIThread.runOnNextRenderFrame(() -> {
                                 UserIdentity value = model.getUserIdentities().get(0);
                                 model.getSelectedUserIdentity().set(value);
-                                updateSaveButtonState();
+                                updateButtons();
                             });
                         }
                     }
Index: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileModel.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileModel.java b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileModel.java
--- a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileModel.java	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileModel.java	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -42,4 +42,6 @@
     private final ObjectProperty<ReputationScore> reputationScore = new SimpleObjectProperty<>();
     private final StringProperty profileAge = new SimpleStringProperty();
     private final BooleanProperty saveButtonDisabled = new SimpleBooleanProperty();
+    private final BooleanProperty deleteButtonDisabled = new SimpleBooleanProperty();
+    private final BooleanProperty useDeleteTooltip = new SimpleBooleanProperty();
 }
Index: desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java
--- a/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/desktop/src/main/java/bisq/desktop/main/content/user/user_profile/UserProfileView.java	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -21,6 +21,7 @@
 import bisq.desktop.common.view.View;
 import bisq.desktop.components.containers.Spacer;
 import bisq.desktop.components.controls.AutoCompleteComboBox;
+import bisq.desktop.components.controls.BisqTooltip;
 import bisq.desktop.components.controls.MaterialTextArea;
 import bisq.desktop.components.controls.MaterialTextField;
 import bisq.i18n.Res;
@@ -28,6 +29,7 @@
 import javafx.geometry.Insets;
 import javafx.geometry.Pos;
 import javafx.scene.control.Button;
+import javafx.scene.control.SplitPane;
 import javafx.scene.image.ImageView;
 import javafx.scene.layout.HBox;
 import javafx.scene.layout.Priority;
@@ -43,13 +45,15 @@
 
 @Slf4j
 public class UserProfileView extends View<HBox, UserProfileModel, UserProfileController> {
-    private final Button createNewProfileButton, deletedButton, saveButton;
+    private final Button createNewProfileButton, deleteButton, saveButton;
+    private final SplitPane deleteWrapper;
     private final MaterialTextField nymId, profileId, profileAge, reputationScoreField, statement;
     private final ImageView roboIconImageView;
     private final MaterialTextArea terms;
     private final VBox formVBox;
     private final AutoCompleteComboBox<UserIdentity> comboBox;
-    private Subscription reputationScorePin, selectedChatUserIdentityPin;
+    private final BisqTooltip deleteTooltip;
+    private Subscription reputationScorePin, useDeleteTooltipPin, selectedChatUserIdentityPin;
 
     public UserProfileView(UserProfileModel model, UserProfileController controller) {
         super(new HBox(20), model, controller);
@@ -110,11 +114,12 @@
         saveButton = new Button(Res.get("action.save"));
         saveButton.setDefaultButton(true);
 
-        deletedButton = new Button(Res.get("user.userProfile.deleteProfile"));
+        deleteButton = new Button(Res.get("user.userProfile.deleteProfile"));
+        deleteWrapper = new SplitPane(deleteButton);
+        deleteTooltip = new BisqTooltip(Res.get("user.userProfile.deleteProfile.tooltip.cannotDelete"));
+        deleteTooltip.getStyleClass().add("medium-dark-tooltip");
 
-        // todo For now we hide the deletedButton because it comes with some complexities
-        // See https://github.com/bisq-network/bisq2/issues/794
-        HBox buttonsHBox = new HBox(20, saveButton/*, deletedButton*/);
+        HBox buttonsHBox = new HBox(20, saveButton, deleteWrapper);
         formVBox.getChildren().add(buttonsHBox);
     }
 
@@ -128,14 +133,18 @@
         terms.textProperty().bindBidirectional(model.getTerms());
         roboIconImageView.imageProperty().bind(model.getRoboHash());
         saveButton.disableProperty().bind(model.getSaveButtonDisabled());
+        deleteButton.disableProperty().bind(model.getDeleteButtonDisabled());
 
+        useDeleteTooltipPin = EasyBind.subscribe(model.getUseDeleteTooltip(), useDeleteTooltip -> {
+            deleteWrapper.setTooltip(useDeleteTooltip ? deleteTooltip : null);
+        });
         reputationScorePin = EasyBind.subscribe(model.getReputationScore(), reputationScore -> {
             if (reputationScore != null) {
                 reputationScoreField.setIconTooltip(reputationScore.getTooltipString());
             }
         });
 
-        deletedButton.setOnAction(e -> controller.onDelete());
+        deleteButton.setOnAction(e -> controller.onDeleteProfile());
         saveButton.setOnAction(e -> controller.onSave());
         createNewProfileButton.setOnAction(e -> controller.onAddNewChatUser());
         comboBox.setOnChangeConfirmed(e -> {
@@ -160,11 +169,14 @@
         terms.textProperty().unbindBidirectional(model.getTerms());
         roboIconImageView.imageProperty().unbind();
         saveButton.disableProperty().unbind();
+        deleteButton.disableProperty().unbind();
+        deleteWrapper.tooltipProperty().unbind();
 
         reputationScorePin.unsubscribe();
+        useDeleteTooltipPin.unsubscribe();
         selectedChatUserIdentityPin.unsubscribe();
 
-        deletedButton.setOnAction(null);
+        deleteButton.setOnAction(null);
         saveButton.setOnAction(null);
         createNewProfileButton.setOnAction(null);
         comboBox.setOnChangeConfirmed(null);
Index: i18n/src/main/resources/user.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/i18n/src/main/resources/user.properties b/i18n/src/main/resources/user.properties
--- a/i18n/src/main/resources/user.properties	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/i18n/src/main/resources/user.properties	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -53,14 +53,14 @@
 user.userProfile.terms.tooLong=Trade terms must not be longer than {0} characters
 
 user.userProfile.createNewProfile=Create new profile
-user.userProfile.deleteProfile=Delete this profile
-user.userProfile.deleteProfile.warning=Do you really want to delete that user profile? You cannot un-do this operation.
-user.userProfile.deleteProfile.warning.yes=Yes, delete my user profile
-user.userProfile.deleteProfile.lastProfile.warning=Deleting user profile is not permitted. \
-  There need to be at least one user profile set up. Please create first another user profile. Once you have more than one user \
-  profiles you can delete this one.
-
-
+user.userProfile.deleteProfile=Delete profile
+user.userProfile.deleteProfile.popup.warning=Do you really want to delete {0}? You cannot un-do this operation.
+user.userProfile.deleteProfile.popup.warning.yes=Yes, delete profile
+user.userProfile.deleteProfile.tooltip.cannotDelete=Deleting user profile is not permitted\n\n\
+  To delete this profile, first:\n\
+  - Delete all messages posted with this profile\n\
+  - Close all private channels for this profile\n\
+  - Make sure to have at least one more profile
 
 
 ################################################################################
Index: user/src/main/java/bisq/user/identity/UserIdentityService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/user/src/main/java/bisq/user/identity/UserIdentityService.java b/user/src/main/java/bisq/user/identity/UserIdentityService.java
--- a/user/src/main/java/bisq/user/identity/UserIdentityService.java	(revision 4f8f535e947b162d2226a73ac00497ee1e78bb2a)
+++ b/user/src/main/java/bisq/user/identity/UserIdentityService.java	(revision 54e8e0e2d0f186154cb63f1a5fd2fa3974ce7f05)
@@ -215,14 +215,10 @@
                 .thenCompose(result -> networkService.publishAuthenticatedData(newUserProfile, oldIdentity.getNodeIdAndKeyPair().getKeyPair()));
     }
 
-    public CompletableFuture<DataService.BroadCastDataResult> deleteUserProfile(UserIdentity userIdentity) {
-        //todo add more checks if deleting profile is permitted (e.g. not used in trades, PM,...)
-        if (getUserIdentities().size() <= 1) {
-            return CompletableFuture.failedFuture(new RuntimeException("Deleting userProfile is not permitted if we only have one left."));
-        }
-
+    public CompletableFuture<DataService.BroadCastDataResult> deleteUserIdentity(UserIdentity userIdentity) {
         synchronized (lock) {
             getUserIdentities().remove(userIdentity);
+
             getUserIdentities().stream().findAny()
                     .ifPresentOrElse(persistableStore::setSelectedUserIdentity,
                             () -> persistableStore.setSelectedUserIdentity(null));

```

The main thing is to move the logic for permitting removal to `UserProfileController`.
`boolean cannotDelete = chatService.isIdentityUsed(userIdentity) || !userIdentityService.hasMultipleUserIdentities();`

As we have multiple cases where we use combinatory logic of different domains in the Desktop module, I am considering to add a new module for that (probably called BisqEasy). 
In the current modules do no clear case which fits for that. We try to keep the domains isolated and a domain can only manage its child domains.
Having a dedicated module for all that domain logic would avoid code duplication in the controllers and make it easier once we have other applications (Web app, mobile,...).



Other schanges are:
- Concat steams in isIdentityUsed
- Avoid to use UI elements in controller, prefer to keep the in view and use only data to control their behaviour. I know there are cases where I did that as well. Some might be candidates for refactoring....
- Tried to make the popup text a bit more user-friendly (not native English speaker, so feel free to improve again if you are)
-  Other smaller refacotrings